build-system/tasks/helpers.js

```diff
@@ -602,39 +615,28 @@ async function compileJsWithEsbuild(srcDir, srcFilename, destDir, options) {
 /**
  * Minify the code with Terser. Only used by the ESBuild.
  *
- * @param {string} destDir
- * @param {string} destFilename
- * @param {?Object} options
- * @return {!Promise}
+ * @param {string} code
+ * @param {string} map
+ * @return {!Promise<{code: string, map: *, error?: Error}>}
  */
-async function minifyWithTerser(destDir, destFilename, options) {
-  if (!options.minify) {
-    return;
-  }
-
-  const filename = path.join(destDir, destFilename);
+async function minify(code, map) {
   const terserOptions = {
-    mangle: true,
-    compress: true,
+    mangle: {},
+    compress: {
+      passes: 3,
+    },
     output: {
       beautify: !!argv.pretty_print,
-      comments: /\/*/,
       // eslint-disable-next-line google-camelcase/google-camelcase
       keep_quoted_props: true,
+      preamble: ';',
     },
-    sourceMap: true,
+    sourceMap: {content: map},
+    module: !!argv.esm,
   };
-  const minified = await terser.minify(
-    fs.readFileSync(filename, 'utf8'),
-    terserOptions
-  );
-  const remapped = remapping(
-    [minified.map, fs.readFileSync(`${filename}.map`, 'utf8')],
-    () => null,
-    !argv.full_sourcemaps
-  );
-  fs.writeFileSync(filename, minified.code);
-  fs.writeFileSync(`${filename}.map`, remapped.toString());
+
+  const minified = await terser.minify(code, terserOptions);
+  return {code: minified.code ?? '', map: minified.map};
```

Comment:
I assume the nullish coalescing is to satisfy type checking? Could `minified.code` ever realistically be nullish, and if so, are we swallowing an error by defaulting to `''`?

Reply:
Exactly right. I don't think it could be nullish. If you prefer, I could check for nullish higher up and throw in that case? I'm not sure what to do when the typing is bad...maybe cast?

Reply:
Hmm. I only asked the question to make sure we didn't miss an easy trick. I don't actually have any ideas to make this better than it is. My guess is that any meaningful error during `terser.minify` will be thrown, and we'll never actually silently return an empty string. And even if we do, the very next step in our build pipeline will likely error out with a different meaningful error.

TL;DR: Okay to leave the code as is, given that this discussion now exists for posterity 😃